x-pack/plugins/security_solution/scripts/endpoint/common/fleet_services.ts

```diff
@@ -121,11 +121,14 @@ export const waitForHostToEnroll = async (
   let found: Agent | undefined;
 
   while (!found && !hasTimedOut()) {
-    found = await fetchFleetAgents(kbnClient, {
-      perPage: 1,
-      kuery: `(local_metadata.host.hostname.keyword : "${hostname}") and (status:online)`,
-      showInactive: false,
-    }).then((response) => response.items[0]);
+    try {
+      found = await fetchFleetAgents(kbnClient, {
+        perPage: 1,
+        kuery: `(local_metadata.host.hostname.keyword : "${hostname}") and (status:online)`,
+        showInactive: false,
+      }).then((response) => response.items[0]);
+      // eslint-disable-next-line no-empty
+    } catch (e) {}
```

Comment:
there was a case where agents API wasn't ready to take requests and it returned an error, so I catch the error and ignore it to let the retry logic continue

Reply:
yeah, I see this change as a big dangerous just because it will keep running (until timeout) even if errors are being encountered. Any way we can check for the specific error you where getting and only ignore that one? I'm also curious what the error was. If Fleet was not yet ready (started), then we're probably calling these APIs too soon?

Reply:
![Zrzut ekranu 2023-02-14 o 14 52 14](https://user-images.githubusercontent.com/5188868/218763211-40b890b9-1fc4-437d-aea6-fcd422ebb14b.png)
It was the error, but it's hard to reproduce. I'm fine with either removing try/catch or adding an error handling

Reply:
Yeah - that error seems like ES just completely crapped out.

I would rather this change be reverted and you handle the possibly failure in the Cypress test suite code instead.

Reply:
this code is not part of Cypress test suite, it's run in the setup phase, so there is no way to handle failure in Cypress code

Reply:
Are you ok reverting this change then? or maybe copy the method over to the Cypress suite and customize it there?